components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/executors/NotificationExecutor.java

```diff
@@ -0,0 +1,106 @@
+import org.wso2.carbon.apimgt.core.impl.Notifier;
+import org.wso2.carbon.apimgt.core.impl.NotifierConstants;
+import org.wso2.carbon.apimgt.core.template.dto.NotificationDTO;
+import org.wso2.carbon.apimgt.core.util.APIUtils;
+
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+
+/**
+ * Used to execute notification threads.
+ */
+public class NotificationExecutor {
+
+    private static final Log log = LogFactory.getLog(NotificationExecutor.class);
+    ExecutorService executor = Executors.newFixedThreadPool(10);
+
+    /**
+     * Executes the notifer classes in separtate threads.
+     *
+     * @param notificationDTO
+     * @throws org.wso2.carbon.apimgt.core.exception.NotificationException
+     */
+    public void sendAsyncNotifications(NotificationDTO notificationDTO) throws NotificationException {
+
+        String notificationTemplate = new APIMConfigurations().
+                getNotificationConfigurations().getNewVersionNotifierConfiguration().getNotifierTemplate();
+        if (notificationDTO.getType().equalsIgnoreCase(NotifierConstants.NOTIFICATION_TYPE_NEW_VERSION)) {
+            JSONParser parser = new JSONParser();
+            JSONObject notificationConfig;
+            try {
+                notificationConfig = (JSONObject) parser.parse(notificationTemplate);
+            } catch (ParseException e) {
+                throw new NotificationException("Error while Parsing Object", e);
+            }
+            JSONArray notifierArray = (JSONArray) notificationConfig.get(NotifierConstants.NOTIFIER);
+
+            for (Object notifierObject : notifierArray) {
+                JSONObject notifierJson = (JSONObject) notifierObject;
+                String notifierClass = (String) notifierJson.get(NotifierConstants.CLASS);
+
+                Properties prop = notificationDTO.getProperties();
+                prop.putAll(notifierJson);
+                notificationDTO.setProperties(prop);
+
+                //starting Notifier threads
+                if (notifierClass != null && !notifierClass.isEmpty()) {
+
+                    Notifier notifier;
+                    try {
+                        notifier = (Notifier) APIUtils.getClassForName(notifierClass).newInstance();
+                    } catch (InstantiationException e) {
+                        throw new NotificationException("Instantiation Error while Initializing the notifier class", e);
+                    } catch (IllegalAccessException e) {
+                        throw new NotificationException("IllegalAccess Error while Initializing the notifier class", e);
+                    } catch (ClassNotFoundException e) {
+                        throw new NotificationException("ClassNotFound Error while Initializing the notifier class", e);
+                    }
+                    notifier.setNotificationDTO(notificationDTO);
+                    Runnable notificationThread = notifier;
```

Comment:
Since notifiier is aleady a Runnable, can we just use  executor.execute(notifier);